Sources/StreamVideoSwiftUI/Livestreaming/LivestreamPlayer.swift

```diff
@@ -0,0 +1,220 @@
+                                    viewModel: viewModel,
+                                    controlsColor: controlsColor
+                                ) {
+                                    participant.track?.isEnabled = !viewModel.streamPaused
+                                    if !viewModel.streamPaused {
+                                        viewModel.update(controlsShown: false)
+                                    }
+                                } : nil
+                            )
+                        }
+                    }
+
+                    if viewModel.controlsShown || !viewModel.fullScreen {
+                        VStack {
+                            Spacer()
+                            HStack(spacing: 8) {
+                                LiveIndicator(controlsColor: controlsColor)
+                                if viewModel.showParticipantCount {
+                                    LivestreamParticipantsView(
+                                        participantsCount: Int(viewModel.call.state.participantCount)
+                                    )
+                                }
+                                Spacer()
+                                LivestreamButton(
+                                    imageName: !viewModel.muted ? "speaker.wave.2.fill" : "speaker.slash.fill"
+                                ) {
+                                    viewModel.toggleAudioOutput()
+                                }
+                                LivestreamButton(imageName: "viewfinder") {
+                                    viewModel.update(fullScreen: !viewModel.fullScreen)
+                                }
+                            }
+                            .padding()
+                            .background(Color.black.opacity(0.4).edgesIgnoringSafeArea(.all))
+                            .foregroundColor(controlsColor)
+                            .overlay(
+                                LivestreamDurationView(
+                                    duration: viewModel.duration(from: state),
+                                    controlsColor: controlsColor
+                                )
+                            )
+                        }
+                    }
+                }
+                .onChange(of: viewModel.fullScreen) { newValue in
+                    onFullScreenStateChange?(newValue)
+                }
+            }
+        }
+        .onChange(of: state.participants, perform: { newValue in
+            if viewModel.muted && newValue.first?.track != nil {
+                viewModel.muteLivestreamOnJoin()
+            }
+        })
+        .onAppear {
+            viewModel.joinLivestream()
+        }
+        .alert(isPresented: $viewModel.errorAlertShown, content: {
+            return Alert.defaultErrorAlert
+        })
+    }
+}
+
+struct LiveIndicator: View {
+    
+    @Injected(\.colors) var colors
+    var controlsColor: Color
```

Comment:
If we go ahead and actually inject the colors we should also accept the background color 1. for consistency 2. changing the foreground may result in the color not being visible it's close the background

Reply:
good point, I will move it to the palette.

Reply:
done